### ISSUE TITLE
x/tools/go/ssa/interp: tests consistently failing on darwin

### ISSUE THREAD

**Issue by mvdan** [golang/go#23166]
Spotted on a recent x/tools CL: https://storage.googleapis.com/go-build-log/2de98eb8/darwin-amd64-10_11_1d3b7b4f.log

```
[...]
Input: static.go
PASS
Input: callstack.go
panic: runtime error: invalid memory address or nil pointer dereference
FAIL
To trace execution, run:
% go build golang.org/x/tools/cmd/ssadump && ./ssadump -build=C -test -run --interp=T callstack.go

The following tests failed:
	boundmeth.go
	complit.go
	coverage.go
	defer.go
	initorder.go
	range.go
	recover.go
	callstack.go
--- FAIL: TestTestdataFiles (23.62s)
	interp_test.go:231: interp.Interpret([testdata/boundmeth.go]) failed: exit code was 2
	interp_test.go:231: interp.Interpret([testdata/complit.go]) failed: exit code was 2
	interp_test.go:231: interp.Interpret([testdata/coverage.go]) failed: exit code was 2
	interp_test.go:231: interp.Interpret([testdata/defer.go]) failed: exit code was 2
	interp_test.go:231: interp.Interpret([testdata/initorder.go]) failed: exit code was 2
	interp_test.go:231: interp.Interpret([testdata/range.go]) failed: exit code was 2
	interp_test.go:231: interp.Interpret([testdata/recover.go]) failed: exit code was 2
	interp_test.go:231: interp.Interpret([testdata/callstack.go]) failed: exit code was 2
FAIL
FAIL	golang.org/x/tools/go/ssa/interp	24.378s
```

The build dashboard shows that this has been happening since December 10th: https://build.golang.org/?repo=golang.org%2fx%2ftools

It's not immediately obvious if it was a change in this repo or a change in Go that introduced this. Or if something is up with the machines.

/cc @alandonovan @bradfitz

**Comment by bradfitz** [comment]
@alandonovan, please fix soon for Go 1.10.

These tests are super fragile.

**Comment by alandonovan** [comment]
On 18 December 2017 at 13:09, Brad Fitzpatrick <notifications@github.com>
wrote:

> @alandonovan <https://github.com/alandonovan>, please fix soon for Go
> 1.10.
>
> These tests are super fragile.
>

Indeed.  I don't have my Mac machine within me today, but I'll take a look
tonight.

**Comment by bradfitz** [comment]
This continuing to fail is interfering with development of unrelated packages in x/tools.

**Comment by mvdan** [comment]
@bradfitz perhaps those tests could be temporarily disabled on Mac until Alan has time to have a look?

**Comment by bradfitz** [comment]
@mvdan, yup, on it.

**Comment by ianlancetaylor** [comment]
It looks we are skipping the tests for now, so this is presumably not a release blocker for 1.10.  Shifting to 1.11.

**Comment by griesemer** [comment]
Again; not a release-blocker for 1.11.

**Comment by bradfitz** [comment]
I think this was fixed along with #27292 by https://github.com/golang/tools/commit/e779aa49e359eb9a328069406fafdb0d559c205d

I no longer see any darwin reference (no more t.Skip).
